typescript/sdk/src/router/schemas.ts

```diff
@@ -0,0 +1,21 @@
+import { z } from 'zod';
+
+import { OwnableConfigSchema } from '../deploy/schemas.js';
+import { ZHash } from '../index.js';
+import { IsmConfigSchema } from '../ism/schemas.js';
+
+export const ForeignDeploymentConfigSchema = z.object({
+  foreignDeployment: z.string().optional(),
+});
+
+export const MailboxClientConfigSchema = z.object({
+  mailbox: ZHash,
+  hook: ZHash.optional(),
```

Comment:
can we make this HookConfigSchema? or saving that for later?

Reply:
Saving that for later. thought to keep this PR scoped to ISM only. 

@yorhodes can you make a PR for that, if it doesnt exist. or i can do it